server/mscalendar/availability.go

```diff
@@ -138,16 +138,20 @@ func (m *mscalendar) GetAvailabilities(remoteUserID string, scheduleIDs []string
 	return client.GetSchedule(remoteUserID, scheduleIDs, start, end, availabilityTimeWindowSize)
 }
 
-func (m *mscalendar) setStatusFromAvailability(mattermostUserID, currentStatus string, av remote.AvailabilityView) string {
-	currentAvailability := av[0]
+func (m *mscalendar) setStatusFromAvailability(mattermostUserID, currentStatus string, s *remote.ScheduleInformation) string {
+	currentAvailability := s.AvailabilityView[0]
 	u := NewUser(mattermostUserID)
 
 	if !m.ShouldChangeStatus(u) {
 		return "Status not changed by user configuration."
 	}
 
 	if !m.HasAvailabilityChanged(u, currentAvailability) {
-		return "Status not changed because there is no update since last status change."
+		ok, newEventTime := m.HasNewEventStarted(u, s)
```

Comment:
A little verbose, but more obvious

```suggestion
		hasStarted, newEventTime := m.HasNewEventStarted(u, s)
```